react-blog-app/src/components/Article.tsx

```diff
@@ -1,27 +1,37 @@
+import { useHistory } from "react-router";
+import useArticle from "../hooks/useArticle";
 import dateStringifier from "../utils/dateStringifier";
 
-const Article = (props: any) => {
-    return (
-        <article>
-            <hr className="horiz-line" />
-            <div className="post-meta-container">
-                <img src={props.authorPhoto} alt={`${props.authorName}'s avatar`} className="author-avatar" />
-                <div className="author-date-container">
-                    <h3 id="author-name">{props.authorName}</h3>
-                    <span className="post-date">{dateStringifier(props.postDate)}</span>
+const Article = ({ id }:any) => {
+
+    const { article, isLoading, error }:any = useArticle(id);
+    const history = useHistory();
+
+    if (error && error.response.status === 404) {
+        history.replace("/404");
+        return null;
+    } else if (isLoading) {
```

Comment:
you don't need to use `else if` and `else` here and below because you `return` from previous condition
you can use only `if` everywhere
```
if (error && error.response.status === 404) {
  return
}

if (isLoading) {
  return
}

return
```
this structure is more readable and flat

Reply:
That’s a very good point! Thank you!